pinot-common/src/main/java/com/linkedin/pinot/common/utils/request/HavingQueryTree.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.common.utils.request;
+
+import com.linkedin.pinot.common.request.AggregationInfo;
+import com.linkedin.pinot.common.request.FilterOperator;
+import java.util.List;
+
+
+public class HavingQueryTree extends QueryTree {
+  final List<HavingQueryTree> _children;
+  private final AggregationInfo _aggregationInfo;
+
+  public HavingQueryTree(AggregationInfo aggregationInfo, List<String> value, FilterOperator operator,
+      List<HavingQueryTree> children) {
+    this._id = System.identityHashCode(this);
```

Comment:
We removed the tree node `id` field in #1740, can you make sure that your code follows this?

Reply:
done